### PR TITLE
update: add missing exception, required_engine_version, required_plugin_version to -L json output

### PR DESCRIPTION
**Pull request by loresuso**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

> /kind release

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area engine

> /area tests

> /area proposals

> /area CI

**What this PR does / why we need it**:
As per PR title

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update: add missing exception, required_engine_version, required_plugin_version to -L json output
```

### CODE REVIEW

**Review by jasondellaluce — COMMENTED** [review]
Thanks Lore! Two things missing before a LGTM:
1. The block at line 581 needs to be reworked to use `create_formatter(r.source, r.output)`, so that the engine creates the formatter with the wrong source. This is blocked by https://github.com/falcosecurity/libs/pull/1120, otherwise we'd be forced to do dynamic casting which would be gross.
2. When a rule/macro match all the event types, return all the types instead of an empty sequence (makes life sooo much easier on automatic checkers)

**Comment by FedeDP** [comment]
@loresuso can you add a release note?

File: cmake/modules/falcosecurity-libs.cmake
Comment (FedeDP) [inline cmake/modules/falcosecurity-libs.cmake:30]: Can we bump to 0.11.0-rc4 here?

**Review by FedeDP — APPROVED** [review]
/approve

File: userspace/engine/json_evt.h
Comment (loresuso) [reply userspace/engine/json_evt.h:441]: you are right, thanks Luca!

**Review by jasondellaluce — APPROVED** [review]
/approve

LGTM

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**
